requests_toolbelt/multipart/encoder.py

```diff
@@ -568,3 +574,53 @@ def len(self):
 
     def read(self, length=-1):
         return self.fd.read(length)
+
+
+class FileFromURLWrapper(object):
+    """File from URL wrapper.
+
+    The :class:`FileFromURLWrapper` object gives you the ability to stream file
+    from provided URL in chunks by :class:`MultipartEncoder`.
+    Provide a stateless solution for streaming file from one server to another.
+
+    .. code-block:: python
+
+        import requests
+        from requests_toolbelt import MultipartEncoder, FileFromURLWrapper
+
+        encoder = MultipartEncoder(
+            fields={'file': FileFromURLWrapper(url, session=session)})
```

Comment:
Multiple examples in the docstring are fine. So let's do this:

```reStructuredText
You can use the :class:`~requests_toolbelt.multipart.encoder.FileFromURLWrapper` without a session or with a session as demonstated by the examples below:


.. code-block:: python

    # No session
    import requests
    from requests_toolbelt.multipart import encoder

    url = 'https://httpbin.org/image/png'
    streaming_encoder = MultipartEncoder(fields={
        'file': FileFromURLWrapper(url),
    })
    r = requests.post('https://httpbin.org/post', data=streaming_encoder,
                      headers={'Content-Type': streaming_encoder.content_type})


.. code-block:: python
    # using a session
    import requests
    from requests_toolbelt.multipart import encoder

    session = requests.Session()
    url = 'https://httpbin.org/image/png'
    streaming_encoder = MultipartEncoder(fields={
        'file': FileFromURLWrapper(url, session=session),
    })
    r = session.post('https://httpbin.org/post', data=streaming_encoder,
                     headers={'Content-Type': streaming_encoder.content_type})
```

Reply:
Sure thing, I will commit those in a sec. 👍